src/main/java/de/cerus/jdasc/JDASlashCommands.java

```diff
@@ -251,34 +263,52 @@ private JDASlashCommands() {
      * @param command  The command you'd like to create / update
      * @param guildId  The id of the guild that should have access to this command
      * @param listener The interaction listener
-     *
      * @return The command id
      */
     public static CompletableFuture<Long> submitGuildCommand(final ApplicationCommand command,
                                                              final long guildId,
                                                              final ApplicationCommandListener listener) {
-        return discordHttpClient.submitGuildCommand(command, guildId).thenApply(response -> {
-            final JsonObject object;
+
+        if(guildCommands.get(guildId) == null || guildCommands.get(guildId).isEmpty()){
+            Map<Long, ApplicationCommand> commandMap = new HashMap<>();
             try {
-                object = JsonParser.parseString(response.body().string()).getAsJsonObject();
-            } catch (final IOException e) {
-                throw new CompletionException(e);
+                for (ApplicationCommand applicationCommand : getGuildCommands(guildId).get()) {
```

Comment:
Using `join()` instead of `get()` would probably be a better solution because you don't need to catch exceptions

Reply:
ok ill do that instead

Reply:
done :D